packages/notifi-react-card/lib/hooks/useIntercomChat.ts

```diff
@@ -80,55 +84,72 @@ export const useIntercomChat = ({
     if (hasNextPage && atTop && isScrolling && visibleRange.startIndex === 0) {
       getConversationMessages();
     }
-  }, [hasNextPage, atTop, isScrolling, visibleRange.startIndex]);
+  }, [
+    getConversationMessages,
+    hasNextPage,
+    atTop,
+    isScrolling,
+    visibleRange.startIndex,
+  ]);
 
   const conversation = useMemo(() => {
     //put the conversation into message group
-    const getFeed = () => {
-      const messageGroups: ChatMessage[][] = [];
-
-      let messages: ChatMessage[] = [];
-
-      chatMessages?.forEach((message, index) => {
-        const nextMessage = chatMessages[index + 1];
-        const isSameUserId = message?.userId === nextMessage?.userId;
-        const isSameDate =
-          formatConversationDateTimestamp(message?.createdDate) ===
-          formatConversationDateTimestamp(nextMessage?.createdDate);
-        /// timestamp logic
-        messages.unshift(message);
-        if (!isSameUserId || !isSameDate) {
-          messageGroups.unshift(messages);
-          messages = [];
-        }
-      });
-
-      const feedEntries = messageGroups.map((messageGroup): FeedEntry => {
-        const firstMessage = messageGroup[0];
-
-        return {
-          direction:
-            firstMessage?.userId === 'c85e8969-10df-43c6-aa4d-402c068e0159'
-              ? 'OUTGOING'
-              : 'INCOMING',
-          id: firstMessage?.id,
-          messages: [...messageGroup],
-          timestamp: firstMessage?.createdDate,
-          type: 'MESSAGES_BLOCK',
-        };
-      });
-
-      return feedEntries;
-    };
+    const messageGroups: ChatMessage[][] = [];
+
+    let messages: ChatMessage[] = [];
+
+    chatMessages?.forEach((message, index) => {
+      const nextMessage = chatMessages[index + 1];
+      const isSameUserId = message?.userId === nextMessage?.userId;
+      const isSameDate =
+        formatConversationDateTimestamp(message?.createdDate) ===
+        formatConversationDateTimestamp(nextMessage?.createdDate);
+      messages.unshift(message);
+      if (!isSameUserId || !isSameDate) {
+        messageGroups.unshift(messages);
+        messages = [];
+      }
+    });
+
+    const feed = messageGroups.map((messageGroup): FeedEntry => {
+      const firstMessage = messageGroup[0];
+
+      return {
+        direction: firstMessage?.userId === userId ? 'OUTGOING' : 'INCOMING',
+        id: firstMessage?.id,
+        messages: messageGroup,
+        timestamp: firstMessage?.createdDate,
+        type: 'MESSAGES_BLOCK',
+      };
+    });
+
     return {
-      feed: getFeed(),
+      feed,
       createdDate:
         chatMessages?.[chatMessages.length - 1]?.createdDate ??
         new Date().toISOString(),
       lastMessage: chatMessages?.[0],
     };
   }, [chatMessages, conversationId]);
 
+  const sendConversationMessages = useCallback(
+    (message: string) => {
+      if (message) {
```

Comment:
In this case, I think that you will always get a message because message always has to be a string and if not optional. 

In this case, I would recommend returning early if the message is not what you wanted. 

Perhaps if message == '' return or throw some error

However, this message may change if you consider my feedback above.